OAuth/ResourceOwner/TwitterResourceOwner.php

```diff
@@ -43,13 +43,23 @@ protected function configureOptions(OptionsResolverInterface $resolver)
             'access_token_url'  => 'https://api.twitter.com/oauth/access_token',
             'infos_url'         => 'https://api.twitter.com/1.1/account/verify_credentials.json',
         ));
-        
-        $resolver->setOptional(array(
-            'x_auth_access_type',
-        ));
-        $resolver->setAllowedValues(array(
+
+        // Symfony <2.6 BC
+        if (method_exists($resolver, 'setDefined')) {
+            $resolver->setDefined('x_auth_access_type');
+        } else {
+            $resolver->setOptional(array(
+                'x_auth_access_type',
+            ));
+        }
+
+        if (method_exists($resolver, 'setDefined')) {
```

Comment:
This is not. I check for `setDefined` method to avoid version_compare and because deprecation on `setAllowedValues` is about arguments, not method itself.

BTW I have to apply it to replace trailing `version_compare`.

Stay tuned.